File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        //      Solution 1
        // for (int i = 0; i < nums.size(); i++)
        //     for (int m = i + 1; m < nums.size(); m++)
        //         if (nums[i] + nums[m] == target)
        //             return {i, m};
        // return {};

        //      Solution 2
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); i++)
        {
            if (m.find(target - nums[i]) != m.end())
                return {m[target - nums[i]], i};
            else
                m[nums[i]] = i;
        }
        return {};
    }
};
// @lc code=end

File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic={}
        for i in range(len(nums)):
            if nums[i] in dic:
                return [dic[nums[i]],i]
            else:
                dic[target-nums[i]]=i
        
        return [-1,-1]   
# @lc code=end

